tutorials/08-voting/contracts/VotingTutorialAdministration.cdc

```diff
@@ -0,0 +1,139 @@
+/*
+*
+*   In this example, we want to create a simple voting contract
+*   where a polling place issues ballots to addresses.
+*
+*   The run a vote, the Admin deploys the smart contract,
+*   then initializes the proposals.
+*   The array of proposals cannot be modified after it has been initialized.
+*
+*   Users can create ballots and vote only with their VotingTutorialGovernanceToken balance prior to when
+*   proposal was created.
+*
+*   Every user with a ballot is allowed to approve their chosen proposals.
+*   A user can choose their votes and cast them
+*   with the tx_05_SelectAndCastVotes.cdc transaction.
+*
+*/
+
+import VotingTutorialGovernanceToken from "./VotingTutorialGovernanceToken.cdc"
+
+pub contract VotingTutorialAdministration {
+
+    // dictionary of proposals to be approved
+    pub var proposals: {Int : ProposalData}
+
+    // paths
+    pub let adminStoragePath: StoragePath
+    pub let ballotStoragePath: StoragePath
+
+    pub struct ProposalData {
+        // the name of the proposal
+        pub let name: String
+        // possible options
+        pub let options: [String]
+        // when the proposal was created
+        pub let blockTs: UFix64
+        // the total votes per option, as represented by the accumulated balances of voters
+        pub(set) var votes: {Int : UFix64}
+        // used to record if a voter as represented by the vault id has already voted
+        pub(set) var voters: {UInt64: Bool}
+
+        init(name: String, options: [String], blockTs: UFix64) {
+            self.name = name
+            self.options = options
+            self.blockTs = blockTs
+            self.votes = {}
+            for index, option in options {
+                self.votes[index] = 0.0
```

Comment:
Do we need this?

Reply:
Yes, because the optional value is force unwrapped in line 99: https://github.com/gus-labs/flow-playground-tutorials/blob/18253165fbaaadc29421acc026b9eee3d6291180/tutorials/08-voting/contracts/VotingTutorialAdministration.cdc#L99